index.d.ts

```diff
@@ -127,9 +127,9 @@ export interface FastifyDynamicSwaggerOptions extends FastifySwaggerOptions {
    */
   stripBasePath?: boolean;
   /**
-   * Overwrite the route schema
+   * custom function to transform the route's schema and url
    */
-  transform?: Function;
+  transform?: ({schema, url}: {schema: JSONObject, url: string}) => { schema: JSONObject, url: string };
```

Comment:
The input is not restricted to JSON object. It can be anything as it is input from user.
```suggestion
  transform?: <S = any>({schema, url}: {schema: S, url: string}) => { schema: JSONObject, url: string };
```

Reply:
done, I didn't commit ur suggestion directly because `JSONObject` was still on the response